common/io/EPoller.cpp

```diff
@@ -312,7 +312,7 @@ bool EPoller::Poll(TimeoutManager *timeout_manager,
 
   int ms_to_sleep = sleep_interval.InMilliSeconds();
   int ready = epoll_wait(m_epoll_fd, reinterpret_cast<epoll_event*>(&events),
```

Comment:
Could we have a comment explaining this for our future selves (you might be able to word it better):
```suggestion
  // If we haven't been asked to wait as part of the poll interval, then don't wait in the epoll
  int ready = epoll_wait(m_epoll_fd, reinterpret_cast<epoll_event*>(&events),
```